### PR TITLE
[CPU] Support topk sort for int32 directly

### PR DESCRIPTION
**Pull request by xuchen-intel**
### Details:
 - *Topk node support sorting without internal data type conversion between interger and floating point in JIT kernel.*

### Tickets:
 - *93903*

### CODE REVIEW

**Comment by xuchen-intel** [comment]
@dmitry-gorokhov Hi Dmitry, could you please take a look?

File: src/plugins/intel_cpu/src/nodes/topk.cpp [hunk 1]
```diff
@@ -788,20 +782,76 @@ struct jit_uni_topk_kernel_f32 : public jit_uni_topk_kernel, public jit_generato
...
+                if (idx_cmp_flg == _cmp_nle_us) {
+                    uni_vpcmpgtd(vmm_mask, vmm_idx_a, vmm_idx_b);
+                } else {
+                    uni_vpcmpgtd(vmm_mask, vmm_idx_b, vmm_idx_a);
+                }
+            }
         }
     }
 
+    inline void compare_node_xmm(Xmm xmm_val_a, Xmm xmm_idx_a, Xmm xmm_val_b, Xmm xmm_idx_b,
```
[the remark this answers is not in the document]
Comment (xuchen-intel) [reply src/plugins/intel_cpu/src/nodes/topk.cpp]: Fixed. Thanks Dmitry!

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/topk.cpp [hunk 2]
```diff
@@ -148,16 +148,23 @@ class TopKLayerCPUTest : public testing::WithParamInterface<TopKLayerCPUTestPara
...
-                rawBlobDataPtr[i] = static_cast<float>(data[i]);
+            if (netPrecision == ElementType::f32) {
+                auto *rawBlobDataPtr = static_cast<float *>(tensor.data());
+                for (size_t i = 0; i < size; ++i) {
+                    rawBlobDataPtr[i] = static_cast<float>(data[i]);
+                }
+            } else {
+                auto *rawBlobDataPtr = static_cast<int32_t *>(tensor.data());
+                for (size_t i = 0; i < size; ++i) {
+                    rawBlobDataPtr[i] = static_cast<int32_t>(data[i]);
```
[the remark this answers is not in the document]
Comment (xuchen-intel) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/topk.cpp:168]: Good point! Fixed. Deliberately set big int32 numbers which ara not accurately representable in fp32. Checked, issue reproduced beforehand. Thanks Dmitry!

File: src/plugins/intel_cpu/src/nodes/topk.cpp [hunk 3]
```diff
@@ -788,20 +782,76 @@ struct jit_uni_topk_kernel_f32 : public jit_uni_topk_kernel, public jit_generato
...
+            } else {
+                vpcmpd(k_mask, xmm_idx_a, xmm_idx_b, idx_cmp_flg);
+            }
+        } else {
+            if (cmp_val) {
+                if (isFloatCompatible(data_type)) {
+                    uni_vcmpps(xmm_mask, xmm_val_a, xmm_val_b, val_cmp_flg);
+                } else {
+                    if (val_cmp_flg == _cmp_nle_us) {
+                        uni_vpcmpgtd(xmm_mask, xmm_val_a, xmm_val_b);
```
[the remark this answers is not in the document]
Comment (xuchen-intel) [reply src/plugins/intel_cpu/src/nodes/topk.cpp]: Fixed. Thanks Dmitry!

**Comment by xuchen-intel** [comment]
> @xuchen-intel Do you have numbers that shows performance change comparing with fp32 execution?
@dmitry-gorokhov It seems that  the model attached in this CVS-93903 don't contain topk nodes. So I don't have models that have topk node with int32 input data. If you can share the model that originally reproduce this issue, I'll be glad to collect the perf numbers. Thanks Dmitry! BTW, as far as I see it, this is a accuracy issue, there shouldn't be significant perf differences.

**Comment by dmitry-gorokhov** [comment]
> > @xuchen-intel Do you have numbers that shows performance change comparing with fp32 execution?
> > @dmitry-gorokhov It seems that  the model attached in this CVS-93903 don't contain topk nodes. So I don't have models that have topk node with int32 input data. If you can share the model that originally reproduce this issue, I'll be glad to collect the perf numbers. Thanks Dmitry! BTW, as far as I see it, this is a accuracy issue, there shouldn't be significant perf differences.

You can just measure performance on tests with different shapes (just add loop over infer call and timer)

**Comment by xuchen-intel** [comment]
> > > @xuchen-intel Do you have numbers that shows performance change comparing with fp32 execution?
> > > @dmitry-gorokhov It seems that  the model attached in this CVS-93903 don't contain topk nodes. So I don't have models that have topk node with int32 input data. If you can share the model that originally reproduce this issue, I'll be glad to collect the perf numbers. Thanks Dmitry! BTW, as far as I see it, this is a accuracy issue, there shouldn't be significant perf differences.
> 
> You can just measure performance on tests with different shapes (just add loop over infer call and timer)

@dmitry-gorokhov Tests show performance gain. Based on SLT, tested shape (21, 21, 21, 21) with the combination of k = {1, 5, 21}, axis = {0, 1, 2, 3}, totally 12 cases. The maximum,minimum, and GeoMean perf gain of int32 over fp32 are 1.22x, 0.99x and 1.09x, respectively. Thanks Dmitry!

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**